### PR TITLE
DSPDC-510 Various UKBB fixes

### PR DESCRIPTION
**Pull request by danxmoran**
Will comment explanations inline. No blockers here, so OK to take time in reviewing.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -26,7 +26,7 @@ pipeline {
             }
         }
         stage('Publish') {
-            when { branch 'master' }
+            //when { branch 'master' }
```
Comment (danxmoran) [inline Jenkinsfile]: Note to self: revert

File: agents/aws-to-gcp/src/main/scala/org/broadinstitute/transporter/transfer/AwsToGcpRunner.scala [hunk 2]
```diff
@@ -287,51 +287,52 @@ class AwsToGcpRunner(
     path: String,
     contentType: `Content-Type`,
     expectedMd5: Option[String],
-    data: Chunk[Byte]
+    data: Stream[IO, Byte]
```
Comment (danxmoran) [inline agents/aws-to-gcp/src/main/scala/org/broadinstitute/transporter/transfer/AwsToGcpRunner.scala]: The logic here now (mostly) matches the implementation in `monster-storage-libs`. We had hit the same multipart upload bug there, then fixed it, but we never backported the fix.

Note to self: Make a ticket to use `monster-storage-libs` in Transporter and delete these implementations.

File: manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala [hunk 3]
```diff
@@ -71,75 +90,42 @@ class TransferController(
     *
     * Batch requests are ordered by their IDs for now.
     */
-  private def lookupSummaries(
```
Comment (danxmoran) [inline manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala]: This was mostly broken. Rewrote it so it wouldn't try to shove all the logic into a single DB transaction.

File: manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala [hunk 4]
```diff
@@ -60,9 +60,28 @@ class TransferController(
     offset: Long,
     limit: Long,
     newestFirst: Boolean
-  ): IO[List[RequestSummary]] =
-    lookupSummaries(Left((offset, limit)), newestFirst)
-      .transact(dbClient)
+  ): IO[List[RequestSummary]] = {
+    val query = for {
+      ids <- List(
+        Fragment.const(s"select id from $RequestsTable order by received_at"),
```
Comment (benjamincarlin) [inline manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala]: Nitpick: If possible capitalize the keywords for readability and syntax.

File: manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala [hunk 5]
```diff
@@ -71,75 +90,42 @@ class TransferController(
...
-    ).combineAll
-
+  private def lookupSummaries(ids: List[UUID]): ConnectionIO[List[RequestSummary]] =
     for {
-      _ <- createTempTable.update.run
-      _ <- buildPivot.update.run
-      summaries <- getSummaryInfo.query[RequestSummary].to[List]
+      countsPerStatusPerRequest <- List(
+        fr"select r.id as id, t.status as status, count(t.id) as n",
+        fr"from" ++ TransfersJoinTable,
```
Comment (benjamincarlin) [inline manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala]: Same

File: manager/src/test/scala/org/broadinstitute/transporter/transfer/TransferControllerSpec.scala [hunk 6]
```diff
@@ -178,31 +178,60 @@ class TransferControllerSpec extends PostgresSpec with MockFactory with EitherVa
...
+        newestToOldest <- controller.listRequests(
+          offset = 0,
+          limit = 2,
+          newestFirst = true
+        )
+      } yield {
+        oldestToNewest should have length 2
+        newestToOldest should have length 2
+        oldestToNewest.reverse shouldBe newestToOldest
+
```
Comment (benjamincarlin) [inline manager/src/test/scala/org/broadinstitute/transporter/transfer/TransferControllerSpec.scala]: The .reverse in nice to know!

File: manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala (hunk 4 above)
Comment (danxmoran) [reply manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala]: +1 for a separate PR that fixes this in all of Transporter

Note to self: make a ticket to do ^^^

Comment (danxmoran) [reply manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala]: https://broadinstitute.atlassian.net/browse/DSPDC-521

File: agents/aws-to-gcp/src/main/scala/org/broadinstitute/transporter/transfer/AwsToGcpRunner.scala (hunk 2 above)
Comment (danxmoran) [reply agents/aws-to-gcp/src/main/scala/org/broadinstitute/transporter/transfer/AwsToGcpRunner.scala]: https://broadinstitute.atlassian.net/browse/DSPDC-522

Comment (danxmoran) [reply agents/aws-to-gcp/src/main/scala/org/broadinstitute/transporter/transfer/AwsToGcpRunner.scala]: https://broadinstitute.atlassian.net/browse/DSPDC-522

File: manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala (hunk 4 above)
Comment (danxmoran) [reply manager/src/main/scala/org/broadinstitute/transporter/transfer/TransferController.scala]: https://broadinstitute.atlassian.net/browse/DSPDC-521

File: manager/src/main/scala/org/broadinstitute/transporter/transfer/api/TransferDetails.scala [hunk 7]
```diff
@@ -20,6 +21,7 @@ import org.broadinstitute.transporter.transfer.TransferStatus
 case class TransferDetails(
   id: UUID,
   status: TransferStatus,
+  priority: Long,
```
Comment (raaidbroad) [inline manager/src/main/scala/org/broadinstitute/transporter/transfer/api/TransferDetails.scala]: short?

File: manager/src/test/scala/org/broadinstitute/transporter/transfer/TransferControllerSpec.scala (hunk 6 above)
Comment (raaidbroad) [reply manager/src/test/scala/org/broadinstitute/transporter/transfer/TransferControllerSpec.scala]: This is a SNAZZY check

File: manager/src/test/scala/org/broadinstitute/transporter/transfer/TransferControllerSpec.scala [hunk 8]
```diff
@@ -503,13 +538,15 @@ class TransferControllerSpec extends PostgresSpec with MockFactory with EitherVa
...
-                 info = '{}'
+                 info = '{}',
+                 priority = 100
                  where id = $id""".update.run.void.transact(tx)
       updatedInfo <- controller.lookupTransferDetails(request1Id, id)
     } yield {
       initInfo shouldBe TransferDetails(
         id,
         TransferStatus.Pending,
+        0L,
```
Comment (raaidbroad) [inline manager/src/test/scala/org/broadinstitute/transporter/transfer/TransferControllerSpec.scala]: Probs update these priorities to `short` too!

**Pull request merged**